app/helpers/Intersect.js

```diff
@@ -571,3 +572,41 @@ export function disableDebugMode() {
   // Kick off intersection calculations to restore data
   checkIntersect();
 }
+
+async function getPageData(authority, page) {
+  // this function should always return an object.
+  // if there's no data, return an empty object.
+  const cacheKey = authority.internal_id + '|page:' + page.id;
```

Comment:
Can this also check that `page.id` is defined? I'm concerned about misconfigured cursor files with undefined ids mapping to the same cache key

Reply:
So if it's undefined, should we continue to download or skip the page? If so, it looks like the id forms part of the filename, so if the id is undefined, we might not be able to download either. But I can add a check here.

Reply:
>  it looks like the id forms part of the filename
I don't believe that'll be part of the contract for the API across all servers (current example erver implementation is moving to a full file url)

I think treating it as an error and skipping sounds good here